### PR TITLE
add pointrend

### PR DESCRIPTION
**Pull request by wuyefeilin**
add pointrend

### CODE REVIEW

File: pdseg/train.py [hunk 1]
```diff
@@ -454,6 +455,9 @@ def var_shape_matched(var, shape):
 
             if args.do_eval:
                 print("Evaluation start")
+                temp = cfg.batch_size_per_dev
+                cfg.batch_size_per_dev = 1
```
Comment (nepeplwu) [inline pdseg/train.py]: 这里设置为1的用处是？

File: pdseg/train.py [hunk 2]
```diff
@@ -454,6 +455,9 @@ def var_shape_matched(var, shape):
...
+                cfg.BATCH_SIZE = cfg.batch_size_per_dev
```
Comment (nepeplwu) [inline pdseg/train.py]: cfg.BATCH_SIZE没有得到恢复

File: pdseg/loss.py [hunk 3]
```diff
@@ -65,7 +75,8 @@ def softmax_with_loss(logit, label, ignore_mask=None, num_classes=2, weight=None
         weighted_label_one_hot.stop_gradient = True
 
     loss = loss * ignore_mask
-    avg_loss = fluid.layers.mean(loss) / fluid.layers.mean(ignore_mask)
+    avg_loss = fluid.layers.mean(loss) / (
+        fluid.layers.mean(ignore_mask) + 0.00001)
```
Comment (nepeplwu) [inline pdseg/loss.py:79]: 是为了预防ignore_mask都为0的情况吗？这种情况会在什么时候出现？

File: pdseg/utils/config.py [hunk 4]
```diff
@@ -168,7 +168,7 @@
 cfg.TEST.TEST_MODEL = ''
 
 ########################## 模型通用配置 #######################################
-# 模型名称, 已支持deeplabv3p, unet, icnet，pspnet，hrnet
+# 模型名称, 已支持deeplabv3p, unet, icnet，pspnet，hrnet, pointrend
```
Comment (nepeplwu) [inline pdseg/utils/config.py]: 没有一个单独的模型名称叫做pointrend吧

File: pdseg/loss.py (hunk 3 above)
Comment (wuyefeilin) [reply pdseg/loss.py:79]: 在cityscape训练的时候如果crop的所有数值都是255的话会出现

File: pdseg/train.py (hunk 1 above)
Comment (wuyefeilin) [reply pdseg/train.py]: 防止评估时显存不足

File: pdseg/train.py (hunk 2 above)
Comment (wuyefeilin) [reply pdseg/train.py]: cfg.BATCH_SIZE在这之后只会用于评估，恢不恢复没有影响了

File: pdseg/train.py [hunk 5]
```diff
@@ -454,6 +455,8 @@ def var_shape_matched(var, shape):
 
             if args.do_eval:
                 print("Evaluation start")
+                temp = cfg.batch_size_per_dev
```
Comment (nepeplwu) [inline pdseg/train.py:459]: 加上注释合适些
